sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -362,13 +412,16 @@ def fit(self, X, y, sample_weight=None):
             self._clear_state()
 
             # initialize raw_predictions: those are the accumulated values
-            # predicted by the trees for the training data. raw_predictions has
+            # predicted by the trees for the training data. raw_prediction has
             # shape (n_trees_per_iteration, n_samples) where
             # n_trees_per_iterations is n_classes in multiclass classification,
             # else 1.
-            self._baseline_prediction = self._loss.get_baseline_prediction(
-                y_train, sample_weight_train, self.n_trees_per_iteration_
-            )
+            # self._baseline_prediction has shape (n_trees_per_iteration, 1)
+            self._baseline_prediction = np.atleast_1d(
+                self._loss.fit_intercept_only(
+                    y_true=y_train, sample_weight=sample_weight_train
+                )
+            ).reshape((-1, 1))
```

Comment:
**Repost to not get lost**

I think `_loss.fit_intercept_only` returns a `np.float64` or `np.float32`, so the following works:

```python
import numpy as np
np.float64(10).reshape(1, -1)
```

(Looking at the code, its the return of `link.link`, and all the link functions end up returning a "numpy scalar" and not a "Python float")

Reply:
See https://github.com/scikit-learn/scikit-learn/pull/20811/commits/1985c947a97aff7130965c977796347be11399d0. Not sure if I made the docstring of `fit_intercept` better or worse.

Reply:
I think the docstring looks good.